src/modules/AutoinstFunctions.rb

```diff
@@ -113,20 +115,27 @@ def selected_product
     #
     # Evaluate all available base products and returns a list of product.
     # CAUTION: The type of the return values depend of the kind of where
-    # the product information has been read (libzypp, product location).
-    # So the type could be ProductLocation or Product.
+    # the product information has been read (libzypp, or product specs).
+    # So the type could be Product or ProductSpec derived class.
+    #
     # available_base_products_hash could be an alternative for this call.
     #
+    # The behaviour of this method can be affected by the `force_libzypp` attribute.
+    # Check {#reset_product} for further details.
+    #
+    # @return [Array<Y2Packager::Product|Y2Packager::ProductSpec>] List of base products
```

Comment:
do you think that it is possible to unify it to product with using `.to_product` ?

Reply:
That's a good question. I am going to check the usage of these AutoinstFunctions methods. Thanks!